R/getStepwiseDivergence.R

```diff
@@ -42,205 +40,97 @@
-                                        n_dimred)})
-
-    x_one_list <- lapply(seq_along(spl_one), function(i) {
-        x[, spl_one[[i]]]}
-    )
-
-    for(i in seq_along(x_one_list)){
-        colData(x_one_list[[i]])[, name_timedifference] <- NA
-        colData(x_one_list[[i]])[, name_divergence] <- NA
-    }
-
-    # assign the names back to (T)SE objects
-    names(x_more_list) <- names(spl_more)
-    names(x_one_list) <- names(spl_one)
-
-    # put lists together and put them in order
-    whole_x <- do.call(c, list(x_one_list, x_more_list))
-    whole_x <- whole_x[order(as.numeric(names(whole_x)))]
-
-    # Merge the objects back into a single X
-    whole_x <- whole_x[!sapply(whole_x,is.null)]
-
-    # Return the SE elements in a list
-    if (length(whole_x) > 1) {
-        x_new <- mergeSEs(whole_x)
-    } else {
-        x_new <- whole_x[[1]]
-    }
-
-    # Ensure that sample sorting matches between the input and output data
-    inds <- match(x$tmp_sample_identifier_for_getStepwiseDivergence,
-                  x_new$tmp_sample_identifier_for_getStepwiseDivergence)
-    x_new <- x_new[, inds]
-
-    # Add the new fields to colData
-    # Just replace the colData for the original input
-    # colData(xorig) <- colData(x_new)
+NULL
 
-    # Add beta divergence from baseline info; note this has to be a list
-    timevalues <- list(colData(x_new)[, name_timedifference])
-    divergencevalues <- list(colData(x_new)[, name_divergence])
-
-    xorig <- .add_values_to_colData(xorig, timevalues, name_timedifference)
-    xorig <- .add_values_to_colData(xorig, divergencevalues, name_divergence)    
-
-    return(xorig)
-
-}
-
-
-#' @rdname getStepwiseDivergence
+#' @rdname addStepwiseDivergence
 #' @export
-setGeneric("getTimeDivergence", signature = c("x"),
-           function(x, ... )
-             standardGeneric("getTimeDivergence"))
+#' 
+#' @importFrom vegan vegdist
+#' @importFrom SummarizedExperiment assay
+#' @importFrom SummarizedExperiment colData
+#' @importFrom SummarizedExperiment colData<-
+#' @importFrom SingleCellExperiment altExp
+#'
+setGeneric("addStepwiseDivergence", signature = c("x"), function(x, ... )
+  standardGeneric("addStepwiseDivergence"))
 
-#' @rdname getStepwiseDivergence
+#' @rdname addStepwiseDivergence
 #' @export
-setMethod("getTimeDivergence",
-          signature = c(x = "ANY"),
-    function(x, ...){
-
-        .Deprecated( msg = paste0("The name of the function 'getTimeDivergence' is",
-                                  " changed to 'getStepwiseDivergence'. \nPlease use the new",
-                                  " name instead.\n",
-                                  "See help('Deprecated')") )
-
-        getStepwiseDivergence(x, ...)
+setMethod("addStepwiseDivergence", signature = c(x = "ANY"),
+    function(
+        x,
+        group=NULL,
+        time.col,
+        time_interval = 1,
+        name = "time_divergence",
+        name.time = "time_difference",
```

Comment:
I think these are ok because we have used `name` widely

Reply:
Ok to me.

Btw term "time" could be also more generic, like "sorter" (...) because the sorting variable could in principle be something else than time. But perhaps this is not so crucial now.

Reply:
True. I think that for divergence values we have to have `name`. I was also wondering about `name = c("divergence", "time_diff")`, but then the input type differs between this and other functions. --> This is rather easy to change in the future, so we can fix this if we come up with better name.

One good point is that `name.time` is quite intuitive about the usage, `name.sorter` might not be